### PR TITLE
Add the ability to mark applications as still in development.

### PR DESCRIPTION
**Pull request by Acidity**

### CODE REVIEW

File: brave/core/application/controller/browse.py [hunk 1]
```diff
@@ -48,13 +48,13 @@ def index(self):
 class BrowseController(Controller):
     @authorize(authenticated)
     def index(self):
-        records = Application.objects()  # TODO: This needs to be ACL-filtered!
-        
-        # TODO: "Development"-only applications.
+        records = Application.objects(development=False or None)  # TODO: This needs to be ACL-filtered!
```
Comment (jboning) [inline brave/core/application/controller/browse.py]: "development=False or None" ?

File: brave/core/application/form.py [hunk 2]
```diff
@@ -20,6 +19,7 @@ def manage_form(action='/application/manage/'):
                     TextArea('description', L_("Description"), rows="6", class_='input-block-level'),
                     URLField('site', L_("Primary Site"), placeholder="http://", required=True, class_='input-block-level validate'),
                     EmailField('contact', L_("Primary Contact"), placeholder="user@example.com", class_='input-block-level'),
+                    TextField('development', L_("Development"), placeholder="True or False", class_='input-block-level'),
```
Comment (jboning) [inline brave/core/application/form.py:22]: Janky. Can we use a checkbox or something instead?

File: brave/core/application/controller/browse.py [hunk 3]
```diff
@@ -48,13 +48,13 @@ def index(self):
...
+        devRecords = Application.objects(development=True)
         
         return 'brave.core.application.template.list_apps', dict(
                 area = 'apps',
-                records = records
+                records = records,
+                devRecords = devRecords
```
Comment (jboning) [inline brave/core/application/controller/browse.py:57]: nitpick: we usually use underscores here, don't we? (If I'm wrong and this is consistent, then that's fine.)

File: brave/core/application/form.py (hunk 2 above)
Comment (Acidity) [inline brave/core/application/form.py:22]: Read my comments!

Comment (jboning) [inline brave/core/application/form.py:22]: Indeed, whoops. Okay I guess :/

File: brave/core/application/controller/browse.py (hunk 3 above)
Comment (Acidity) [inline brave/core/application/controller/browse.py:57]: hmm, I don't think so, and I was just following the convention of the code immediately surrounding it. the _ are for json dicts with messages I believe.

File: brave/core/application/controller/browse.py (hunk 1 above)
Comment (Acidity) [inline brave/core/application/controller/browse.py]: For compatibility with existing databases, if it was just False then those which had been added before the change would be non-visible to anyone on the available applications page. All new apps should have False by default though.

Comment (jboning) [inline brave/core/application/controller/browse.py]: I see. I don't think this is going to accomplish what you want, though; I think you want `development__in=[False, None]`

Comment (Acidity) [inline brave/core/application/controller/browse.py]: It worked perfectly in testing...

Comment (Acidity) [inline brave/core/application/controller/browse.py]: Just kidding, you're right,

**Pull request merged**